### PR TITLE
Revisions for documentation

### PR DESCRIPTION
**Pull request by jcstein**
- [x] Change intro on 1st paragraph https://github.com/rollkit/docs/pull/57/commits/432692d31008499f20ed0209f4a9950cd81f3071
- [x] update developer pain points https://github.com/rollkit/docs/pull/57/commits/3be9e3ac6a06a232094951d9f4d1f3c458e5cebc
- [x] Remove core concepts https://github.com/rollkit/docs/pull/57/commits/71774dc4e631448b4b9189b3070d5a2d606b644b
- [x] edit rollkit stack / intro https://github.com/rollkit/docs/pull/57/commits/26e7e2a0a017b0a83c86383e6be8409110303ffd
- [x] change tagline and descriptions https://github.com/rollkit/docs/pull/57/commits/ddd0b91cf9c62f076040d27f61cffc01a669810c
- [x] standardize `-replace` https://github.com/rollkit/docs/pull/57/commits/3b9f16fecf36a27f9661f12e05b594fa9459ca10
- [x] remove MD013 linting rule (line length, max. 80 characters) https://github.com/rollkit/docs/pull/57/commits/c60e1ea9a256494e6492b6af6081db3105324af3 https://github.com/rollkit/docs/pull/57/commits/d45757d5f8a7567ebf00b9b1fa217005590a7758
- [x] Get team review
- [ ] Add team revisions

### CODE REVIEW

**Comment by nashqueue** [comment]
@jcstein can you remove core concepts before rereview ?

**Comment by jcstein** [comment]
Yes, done @nashqueue

**Comment by musalbas** [comment]
@jcstein It's better to delete the core concepts page from this review, and submit it a different PR, because that page needs more work, so it shouldn't block the review for the other pages (intro and rollkit stack).

File: docs/rollkit-stack.md [hunk 1]
```diff
@@ -1,30 +1,26 @@
...
 sidebar_label: Rollkit Stack
-description: Intro to Rollkit's stack, a modular development kit for building rollup chains.
+description: Intro to Rollkit's stack, a modular rollup framework.
 ---
 
 # The Rollkit Stack
 
 This section will cover the technical stack of Rollkit.
 
-Rollkit is built by replacing Tendermint, the Cosmos-SDK Consensus Layer,
```
Comment (musalbas) [inline docs/rollkit-stack.md:11]: why remove this? It seems like important context for the rollkit stack

Comment (musalbas) [reply docs/rollkit-stack.md:11]: (also as mentioned in my google doc, it's not accurate to refer to Tendermint as "the Cosmos consensus layer")

Comment (jcstein) [reply docs/rollkit-stack.md:11]: It was duplicated in the intro and rollkit stack

**Comment by jcstein** [comment]
done @musalbas

File: docs/core-concepts.md [hunk 2]
```diff
@@ -129,28 +100,25 @@ set will be able to opt-in to become a validator for other zones.
 
 ## Fraud Proofs
 
-The Rollkit chains will have no consensus rules. All valid transactions will
-be accepted by block creators. Fraud proofs will be the only way to detect
-and rollback malicious state transitions.
+The Rollkit chains will inherit the consensus of the consensus layer when
+it comes to ordering rollup blocks. All valid rollup blocks on the DA layer
+will be accepted by nodes. In the case of a pessimistic rollup, you can
+run a full node for the rollup for fraud proofs. In a zk-rollup, there will
```
Comment (tzdybal) [inline docs/core-concepts.md]: Pessimistic rollups are rollups where each node is full - there is no need for fraud proofs as everyone executes all transactions.
Optimistic rollups require fraud proofs. Full nodes are needed to create fraud proofs.

File: docs/rollkit-stack.md [hunk 3]
```diff
@@ -1,30 +1,26 @@
...
-and adaptability.
-
-It spins up a sovereign rollup, which collects transactions into blocks and
+It spins up a rollup, which collects transactions into blocks and
 posts them onto Celestia for DA and Consensus.
 
-The goal of Rollkit is to enable anyone to design and deploy a sovereign
-rollup on Celestia in minutes with minimal overhead.
+The goal of Rollkit is to enable anyone to design and deploy a
+rollup on any DA layer with minimal overhead. Rollkit aims to be a neutral
```
Comment (tzdybal) [inline docs/rollkit-stack.md:15]: credibly neutral

File: docs/rollkit-stack.md [hunk 4]
```diff
@@ -55,9 +51,9 @@ nonce/sequence number). This behaviour is similar to the Tendermint mempool.
 ### Light node
 
 Light nodes are the main producer of transactions in the Rollkit network.
-They participate in gossiping of and fraud proofs. Light nodes may only
-request or store a subset of the state, just to ensure that they can execute
-rollback.
+They participate in gossiping of fraud proofs. Light nodes may only
+request or store a subset of the state, so they can query chain balances
```
Comment (tzdybal) [inline docs/rollkit-stack.md:55]: We shouldn't put this here. Right now we have no state at all, and in the future, we will work on fancy trust-minimized approach for queries, but probably without explicit state storage.

**Review by tzdybal — COMMENTED** [review]
Left some comments, first pass.

**Comment by gupadhyaya** [comment]
Comments for the landing page:
    - "A modular rollup framework" => "A framework for building modular rollups"
    - "Build Modular" => "Learn More"
    - "Rollups can fork like any L1" => "Freedom to configure the execution environment" 
    - Re-arrange the 4 items to: Secure, Scalable, Sovereign, Easy to Deploy. This makes it easier for read one after the other for me.

I do not like the current structure and content of the intro. the intro must address:
    - what is rollkit
    - why it is needed (what problem it is solving)
    - how it is cool? (how does it solve the challenges)
    - tell me more

what it is?
Rollkit is a framework designed to ease the development and deployment of modular blockchains, specifically the execution layers of the modular blockchain stack (aka rollups) (link to celestia article). 

why it is needed?
Today, the rollups are built 1) on top of monolithic base layer or 2) as independent app-chains. In case of rollups that utilize the monolithic base layer inherit the security from the base layer, however they lack scalability as they have to compete with other apps for base layer's limited block space. Further, they lack the freedom to pick and customize their execution environment (aka sovereignty). On the other hand, the app-chains are sovereign, but have to manage their own security by spinning up and maintaining a decentralized validator network. Further, they are not truely modular as they also perform consensus and settlement.

The rollups built using Rollkit are 1) secured by the base layer (unlike app-chains which requires bootstraping a decentralized set of validators to secure the rollup), 2) scalable due to modular base layer which unlike monolethic base layer, does not incur the problem of competing for limited block space, and 3) sovereign due to their freedom to pick and configure their execution environments.
On top of these key benefits, building and deploying rollups requires minimal development efforts.

<elaborate more on security, scalability, sovereignity, and ease of deloyment factors>

\<envisioned usecases>

tell me more?
<elaborate on how rollkit is built and make it interesting for people to read on to next sections>

i also recommend having a paragraph or bullet points describing the current state of rollkit.

**Comment by gupadhyaya** [comment]
For the Rollkit Stack page, i would structure this way:

For understanding the Rollkit stack, let's first look at the key components of a rollup. A typical rollup has:
    - an rpc layer for exploring the rollup chain and submit transactions
    - a mempool for queing up the transactions
    - a set of nodes of different kinds 
        - sequencer/aggregator nodes that bundles the transactions (rollup) to create a rollup block to be submitted to base layer for data availability
        - full nodes for maintaining and serving the rollup 
        - light nodes that help validate the rollup transactions
    - an executor (or a virtual machine) for transaction execution and state transition
    - a validation mechanism such as fraud proof or validity proofs
    - underlying networking layer such as p2p
    - \<if i have missed any other key component>

\<expand on what Rollkit offers for each of these key component and explain how the flexibilty exists to customize the offerings>

File: docs/intro.md [hunk 5]
```diff
@@ -1,109 +1,36 @@
...
+description: Intro to Rollkit, a modular rollup framework.
 ---
 
 # Introduction to Rollkit
 
-Rollkit is a modular development kit for anyone to easily deploy their own
-blockchain with minimal overhead. It solves the problem of bootstrapping a
-secure and decentralized validator set while removing the burden of maintaining
-a new consensus network.
+Rollkit is a rollup framework that gives developers the freedom to
```
Comment (AlCJoseph) [inline docs/intro.md:9]: "Rollkit is a modular rollup framework that gives developers the freedom to deploy throughout the modular stack, opening new possibilities of rapid experimentation and innovation"

File: docs/core-concepts.md [hunk 6]
```diff
@@ -16,28 +16,16 @@ If you're familiar with Rollkit's core concepts, you may want to skip to the
...
-scalability of blockchain networks and application-specific blockchains,
-often known as rollups, allowing them to process more transactions and
-support a larger number of users.
+A rollup is a blockchain that uses another chain as its consensus and data
+availability (DA) layer, but does execution off-chain to the consensus and
+DA chain. The rollup inherits security from the consensus and DA layer, by
+making transactions available for fraud proofs and optimistic rollups, or
+making state accessible for zero knowledge (ZK) rollups. Rollups are able
+to implement unique execution environments without needing a new Layer 1
+blockchain, such as Fuelmint, CosmWasm, and Ethermint. Scalability for
```
[the remark this answers is not in the document]
Comment (jcstein) [reply docs/core-concepts.md]: yes, great idea

[the remark this answers is not in the document]
Comment (jcstein) [reply docs/core-concepts.md]: however, have removed core concepts on this PR

File: docs/core-concepts.md (hunk 2 above)
Comment (jcstein) [reply docs/core-concepts.md]: copy that. thank you for this review Tomasz. Will implement in PR for core concepts

File: docs/intro.md (hunk 5 above)
Comment (S1nus) [reply docs/intro.md:9]: we don't need "modular" twice in one sentence

File: docs/rollkit-stack.md (hunk 3 above)
Comment (jcstein) [reply docs/rollkit-stack.md]: resolved in https://github.com/rollkit/docs/pull/57/commits/eb6729eb1217ff7d2bea3d586c0909ae5032c99e

File: docs/rollkit-stack.md (hunk 4 above)
Comment (jcstein) [reply docs/rollkit-stack.md:55]: should we remove light node for now?

Comment (nashqueue) [reply docs/rollkit-stack.md:43]: No light nodes should be included. The statement is false.

Comment (musalbas) [reply docs/rollkit-stack.md:43]: Could also be included somehow in such a way that describes it as a future feature / work in progress.

Comment (jcstein) [reply docs/rollkit-stack.md:43]: @S1nus will expand on light nodes

**Pull request merged**